test/naming_tests.jl

```diff
@@ -80,3 +80,22 @@ x = A(5, NamedTuple())
 # container with values that are not by default printed in savename
 x = A(5, (m = rand(50,50),))
 @test !occursin("p", savename(x))
+
+# Scientific notation for savename
+a = 1.2345e-7
+b = 1.0
+c = 1
+d = "test"
+di = @dict a b c d
+
+@test savename(di,scientific=6) == "a=1.2345e-7_b=1.0_c=1_d=test"
+@test savename(di,scientific=5) == "a=1.2345e-7_b=1.0_c=1_d=test"
+@test savename(di,scientific=4) == "a=1.234e-7_b=1.0_c=1_d=test"
+@test savename(di,scientific=3) == "a=1.23e-7_b=1.0_c=1_d=test"
+@test savename(di,scientific=2) == "a=1.2e-7_b=1.0_c=1_d=test"
+@test savename(di,scientific=1) == "a=1.0e-7_b=1.0_c=1_d=test"
```

Comment:
In the case of `scientific = 1`, isn't the second digit always `0` and thus can be skipped?

Reply:
Yes it is. Do you think it's better to fallback to the rounding behavior of the default rounding method? So also adding the check with Integer rounding.

Reply:
In favor of consistency it's better to remove it

Reply:
> Yes it is. Do you think it's better to fallback to the rounding behavior of the default rounding method?

Yes I think so.

Reply:
(to be clear I think that `1.0e-7` should be written as `1e-7`. But if this is too hard to do programatically I can live with it not happening.

Reply:
Ah, now I get it. That shouldn't be too difficult.